### PR TITLE
latex printing of Cycle, Permutation

### LINKED ISSUE #10210: LaTex printing of Cycle
Currently
```
>>> from sympy.combinatorics.permutations import Cycle
>>> latex(Cycle(1, 2))
'\\left \\{ 1 : 2, \\quad 2 : 1\\right \\}'
```
That's something of latex for dictionary(`dict`). The LaTex should give the latex representation of `(1 2)`. i.e `latex(Cycle(1, 2))` should return the latex representation of `(1 2)`.

In IPython Notebook
```
>>> init_printing(use_unicode=True)
>>> Cycle(1, 2)
{0:0,1:2,2:1}
```
You can also see the discussion [here](https://github.com/sympy/sympy/pull/10183#issuecomment-162313211) on PR #10183 . Changes should be made to file `printing/latex.py`(mentioned by jksuom) This will be an "easy to fix" issue.

### PR DESCRIPTION
**Pull request by gxyd**
Now

```
>>> init_printing(use_unicode=True)
>>> Cycle(1, 2, 4)(5, 6)
(1 2 4)(5 6)
>>> latex(Cycle(1, 2, 4)(5, 6))
'\\left( 1\\, 2\\, 4\\right)\\left( 5\\, 6\\right)'
```

same printing for Permutation

fixes #10210

### CODE REVIEW

File: sympy/printing/tests/test_latex.py [hunk 1]
```diff
@@ -128,6 +129,17 @@ def test_latex_builtins():
     assert latex(false) == r'\mathrm{False}'
 
 
+def test_latex_cycle():
+    assert latex(Cycle(1, 2, 4)) == r"\left( 1\, 2\, 4\right)"
+    assert latex(Cycle(1, 2)(4, 5)) == r"\left( 1\, 2\right)\left( 4\, 5\right)"
+    assert latex(Cycle()) == r"\left( \right)"
+
+
+def tes_latex_permutation():
```
Comment (scopatz) [inline sympy/printing/tests/test_latex.py]: shouldn't this be `test`

**Comment by jksuom** [comment]
I have not studied how this would appear on screen or paper, but does the ``\thinspace`` (``\,``) suffice to separate the numbers visually? Perhaps you could also test with ``\:`` or ``\;`` (``\medspace`` or ``\thickspace``).

**Comment by gxyd** [comment]
> does the \thinspace (\,) suffice to separate the numbers visually

Yes i think it does.
Here are the few tests [sympy1.pdf](https://github.com/sympy/sympy/files/62229/sympy1.pdf) 
In order with "/,"   ,   "/:"    and  "/;"   respectively.

**Comment by jksuom** [comment]
I am not quite convinced by these examples that involve only single digits. What would happen with two-digit numbers? (The line with ``\medspace`` looks also reasonable.)

**Comment by gxyd** [comment]
Here are some screenshots. I don't think my `LaTex` installation supports `\medspace` or `thickspace`. I tried with spacings mentioned on [this answer](http://tex.stackexchange.com/a/74354/88362) .
screen_sympy_printing

I think i like the `\enspace` printing. Since it ensures the adequate spacing.

**Comment by jksuom** [comment]
``\enspace`` looks about twice the size of ``\medmuskip`` (given by ``\:`` or ``\>``) on my screen. It seems to me that it is more than adequate. Perhaps ``\medmuskip`` or ``\thickmuskip`` would suffice.

**Comment by gxyd** [comment]
I don't know how the `\thickmuskip` looks on printing(seems like my `LaTex`distribution does not support it). Though i have changed it to `\thickmuskip`. Plese have a look.

File: sympy/printing/latex.py [hunk 2]
```diff
@@ -280,6 +280,19 @@ def _print_Add(self, expr, order=None):
...
         return tex
 
+    def _print_Cycle(self, expr):
+        from sympy.combinatorics.permutations import Permutation, Cycle
+        if str(expr) == '()':
+            return r"\left( \right)"
+        expr_perm = Permutation(expr).cyclic_form
+        term_tex = ''
+        for i in expr_perm:
+            term_tex += str(i).replace(',', r"\thickmuskip")
```
Comment (jksuom) [inline sympy/printing/latex.py]: This should be ``\;`` to actually create the space. (My use of the name of space may have been misleading, I'm sorry.)

**Comment by jksuom** [comment]
``\thinmuskip`` (thin math space), ``\medmuskip`` (medium math space) and ``\thickmuskip`` (thick math space) denote the *size* of the skip (they do not create it). They are defined in the TeXbook as follows.
``\thinskip=3mu``
``\medskip=4mu plus 2mu minus 4mu``
``\thickskip=5mu plus 5mu``
Here ``mu`` (math unit) is a unit of length depending on the font in use. (I think that it is usually one eighteenth of ``\em`` that is the size of the letter "m" in the font. The size of ``\enskip`` is one half ``\em``.)
``\medskip`` can stretch of shrink to adjust for the available space. It can vanish altogether. Hence it may be preferable to use ``\thickskip``.

These spaces do not appear explicitly in latex text. Instead, they are invoked by the commands ``\,``, ``\>`` and ``\;`` that are defined as follows (``\mskip`` will create the space).
``\def\,{\mskip\thinskip}``
``\def\>{\mskip\medskip}``
``\def\;{\mskip\thickskip}``

**Comment by gxyd** [comment]
Thanks for the detailed explanation. I am learning `LaTex` as well these days, though currently only know a few basics. I didn't knew these wide variety of horizontal space commands.
Though for the PR, i think is understood you comment and have made the changes. I think they also work ok on my system.

**Comment by jksuom** [comment]
Looks good. +1

**Comment by jksuom** [comment]
Merging in a few hours if there are no comments.

**Comment by gxyd** [comment]
Thanks @jksuom @certik :)

**Pull request merged**